aspnetcore/fundamentals/middleware.md

```diff
@@ -188,18 +188,22 @@ app.Map("/level1/level2", HandleMultiSeg);
 
 ## Built-in middleware
 
-ASP.NET Core ships with the following middleware components:
-
-| Middleware | Description |
-| ----- | ------- |
-| [Authentication](xref:security/authentication/identity) | Provides authentication support. |
-| [CORS](xref:security/cors) | Configures Cross-Origin Resource Sharing. |
-| [Response Caching](xref:performance/caching/middleware) | Provides support for caching responses. |
-| [Response Compression](xref:performance/response-compression) | Provides support for compressing responses. |
-| [Routing](xref:fundamentals/routing) | Defines and constrains request routes. |
-| [Session](xref:fundamentals/app-state) | Provides support for managing user sessions. |
-| [Static Files](xref:fundamentals/static-files) | Provides support for serving static files and directory browsing. |
-| [URL Rewriting Middleware](xref:fundamentals/url-rewriting) | Provides support for rewriting URLs and redirecting requests. |
+ASP.NET Core ships with the following middleware components, as well as a description of the order in which they run:
+
+| Middleware | Description | Ordering |
+| ----- | ------- | ------- |
+| [Authentication](xref:security/authentication/identity) | Provides authentication support. | Before `HttpContext.User` is needed. Terminal for OAuth callbacks. |
+| [CORS](xref:security/cors) | Configures Cross-Origin Resource Sharing. | Before components that use CORS |
+| [Response Caching](xref:performance/caching/middleware) | Provides support for caching responses. | Before components that want caching |
+| [Response Compression](xref:performance/response-compression) | Provides support for compressing responses. | Before components that want to use compression |
+| [Routing](xref:fundamentals/routing) | Defines and constrains request routes. | Terminal for matching routes. |
+| [Session](xref:fundamentals/app-state) | Provides support for managing user sessions. | Before components that want to use Session. |
+| [Static Files](xref:fundamentals/static-files) | Provides support for serving static files and directory browsing. | Terminal if a request matches files |
+| [URL Rewriting ](xref:fundamentals/url-rewriting) | Provides support for rewriting URLs and redirecting requests. | Before components that consume the URL. |
+| [Diagnostics](xref:security/) | Configure diagnostics. |  |
+| [HttpOverrides/ForwardedHeaders](xref:security/) | x. |  |
```

Comment:
Before components that consume the fields being updated (e.g. Scheme, Host, ClientIP, Method)